File: app/src/main/java/com/example/zebrapoc/broadcastReceiver/PowerConnectionReceiver.java
package com.example.zebrapoc.broadcastReceiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.widget.Toast;

import androidx.core.content.ContextCompat;

import com.example.zebrapoc.service.EventTrackingService;
import com.example.zebrapoc.utils.ExportFile;

public class PowerConnectionReceiver extends BroadcastReceiver {

    public PowerConnectionReceiver() {
    }

    @Override
    public void onReceive(Context context, Intent intent) {

        if (intent.getAction().equals(Intent.ACTION_POWER_CONNECTED)) {
            Toast.makeText(context, "POWER CONNECTED", Toast.LENGTH_SHORT).show();
            //Intent serviceIntent = new Intent(context, EventTrackingService.class);
            //context.stopService(serviceIntent);

            ExportFile.exportDataIntoCSVFile(context, "onPowerConnected");
        }
        if (intent.getAction().equals(Intent.ACTION_POWER_DISCONNECTED)) {
            Toast.makeText(context, "POWER DISCONNECTED", Toast.LENGTH_SHORT).show();
            //Intent serviceIntent = new Intent(context, EventTrackingService.class);
            //serviceIntent.putExtra("inputExtra", "Accelerometer running");
            //ContextCompat.startForegroundService(context, serviceIntent);
        }
        if (intent.getAction().equals(Intent.ACTION_BATTERY_LOW)) {
            Toast.makeText(context, "BATTERY LOW", Toast.LENGTH_LONG).show();
        }
        if (intent.getAction().equals(Intent.ACTION_BATTERY_OKAY)) {
            Toast.makeText(context, "BATTERY OKAY", Toast.LENGTH_LONG).show();
        }

        //Calling repeatedly
        /* if (intent.getAction().equals(Intent.ACTION_BATTERY_CHANGED)) {
            Toast.makeText(context, "BATTERY CHANGED", Toast.LENGTH_LONG).show();
        }*/
    }
}

File: app/src/main/java/com/alumnus/zebra/broadcastReceiver/PowerConnectionReceiver.kt
package com.alumnus.zebra.broadcastReceiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import android.widget.Toast
import androidx.core.content.ContextCompat
import com.alumnus.zebra.machineLearning.utils.ExportFiles.prepareDataChunk
import com.alumnus.zebra.service.LifeTimeService

class PowerConnectionReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == Intent.ACTION_POWER_CONNECTED) {
            Toast.makeText(context, "POWER CONNECTED", Toast.LENGTH_SHORT).show()

            //region Stop service on power connect
            /*val serviceIntent = Intent(context, LifeTimeService::class.java)
            context.stopService(serviceIntent)*/
            //endregion
            prepareDataChunk(context, false)
        }
        if (intent.action == Intent.ACTION_POWER_DISCONNECTED) {
            Toast.makeText(context, "POWER DISCONNECTED", Toast.LENGTH_SHORT).show()

            //region Start service on power disconnect
            val sp = context.getSharedPreferences("Zebra", Context.MODE_PRIVATE)
            val frequency = sp.getInt("frequency", 5)
            val serviceIntent = Intent(context, LifeTimeService::class.java)
            serviceIntent.putExtra("frequency", frequency)
            /*if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O)
                ContextCompat.startForegroundService(context, serviceIntent)
            else context.startService(intent)*/

            //endregion
        }
        if (intent.action == Intent.ACTION_BATTERY_LOW) {
            Toast.makeText(context, "BATTERY LOW", Toast.LENGTH_LONG).show()
        }
        if (intent.action == Intent.ACTION_BATTERY_OKAY) {
            Toast.makeText(context, "BATTERY OKAY", Toast.LENGTH_LONG).show()
        }

        //Calling repeatedly
        /*if (intent.action == Intent.ACTION_BATTERY_CHANGED) {
            Toast.makeText(context, "BATTERY CHANGED", Toast.LENGTH_LONG).show()
        }*/
    }
}